File: create_icon.py
#!/usr/bin/env python3

import os
from PIL import Image, ImageDraw, ImageFont
import json

def create_app_icon():
    """Create a simple gradient app icon with 'R' letter"""
    
    # Create 1024x1024 image with gradient
    size = 1024
    img = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    draw = ImageDraw.Draw(img)
    
    # Create aurora gradient background
    for y in range(size):
        # Gradient from purple to blue
        progress = y / size
        r = int(100 * (1 - progress) + 40 * progress)
        g = int(50 * (1 - progress) + 100 * progress)
        b = int(255 * (1 - progress) + 255 * progress)
        draw.rectangle([(0, y), (size, y+1)], fill=(r, g, b, 255))
    
    # Add subtle radial gradient overlay
    for i in range(size//4, 0, -1):
        alpha = int(255 * (1 - i/(size//4)) * 0.3)
        color = (255, 255, 255, alpha)
        draw.ellipse(
            [(size//2 - i, size//2 - i), (size//2 + i, size//2 + i)],
            fill=None,
            outline=color,
            width=2
        )
    
    # Draw "R" letter in center
    try:
        # Try to use a system font
        font_size = size // 2
        font = ImageFont.truetype("/System/Library/Fonts/Helvetica.ttc", font_size)
    except:
        # Fallback to default font
        font = ImageFont.load_default()
    
    text = "R"
    # Get text bbox
    bbox = draw.textbbox((0, 0), text, font=font)
    text_width = bbox[2] - bbox[0]
    text_height = bbox[3] - bbox[1]
    
    # Center the text
    x = (size - text_width) // 2
    y = (size - text_height) // 2 - text_height // 10
    
    # Draw text with glow effect
    for offset in range(10, 0, -1):
        alpha = int(255 * (1 - offset/10) * 0.5)
        draw.text((x, y), text, font=font, fill=(255, 255, 255, alpha))
    
    # Draw main text
    draw.text((x, y), text, font=font, fill=(255, 255, 255, 255))
    
    # Save the icon
    icon_path = "/Users/alisimacpro/Desktop/RayLink/RayLink/Assets.xcassets/AppIcon.appiconset/AppIcon.png"
    img.save(icon_path, "PNG")
    print(f"✅ Created app icon: {icon_path}")
    
    # Update Contents.json to reference the icon
    contents_path = "/Users/alisimacpro/Desktop/RayLink/RayLink/Assets.xcassets/AppIcon.appiconset/Contents.json"
    contents = {
        "images": [
            {
                "filename": "AppIcon.png",
                "idiom": "universal",
                "platform": "ios",
                "size": "1024x1024"
            }
        ],
        "info": {
            "author": "xcode",
            "version": 1
        }
    }
    
    with open(contents_path, 'w') as f:
        json.dump(contents, f, indent=2)
    print(f"✅ Updated Contents.json")

if __name__ == "__main__":
    try:
        from PIL import Image, ImageDraw, ImageFont
        create_app_icon()
    except ImportError:
        print("⚠️  PIL not installed. Using fallback method...")
        # Create a simple 1x1 PNG as placeholder
        import base64
        
        # 1x1 blue PNG
        png_data = base64.b64decode(
            b'iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mNkYPhfDwAChwGA60e6kgAAAABJRU5ErkJggg=='
        )
        
        icon_path = "/Users/alisimacpro/Desktop/RayLink/RayLink/Assets.xcassets/AppIcon.appiconset/AppIcon.png"
        with open(icon_path, 'wb') as f:
            f.write(png_data)
        print("✅ Created placeholder app icon")

File: create_icon.swift
#!/usr/bin/swift

import Foundation
import CoreGraphics
import UniformTypeIdentifiers

// Create a simple gradient app icon
func createAppIcon(size: Int, path: String) {
    let width = size
    let height = size
    let bytesPerPixel = 4
    let bytesPerRow = bytesPerPixel * width
    let bitsPerComponent = 8
    
    var pixelData = [UInt8](repeating: 0, count: width * height * bytesPerPixel)
    
    // Create aurora gradient
    for y in 0..<height {
        for x in 0..<width {
            let offset = (y * width + x) * bytesPerPixel
            
            // Create gradient from purple to blue
            let progress = Double(y) / Double(height)
            let r = UInt8(128 * (1 - progress) + 64 * progress)
            let g = UInt8(64 * (1 - progress) + 128 * progress)
            let b = UInt8(255 * (1 - progress) + 255 * progress)
            
            pixelData[offset] = r     // R
            pixelData[offset + 1] = g // G
            pixelData[offset + 2] = b // B
            pixelData[offset + 3] = 255 // A
        }
    }
    
    // Add "R" letter in center
    let centerX = width / 2
    let centerY = height / 2
    let letterSize = width / 3
    
    for y in (centerY - letterSize/2)..<(centerY + letterSize/2) {
        for x in (centerX - letterSize/3)..<(centerX + letterSize/3) {
            let offset = (y * width + x) * bytesPerPixel
            
            // Simple "R" shape
            let relX = x - (centerX - letterSize/3)
            let relY = y - (centerY - letterSize/2)
            
            var isLetter = false
            
            // Vertical line of R
            if relX < letterSize/8 {
                isLetter = true
            }
            // Top curve of R
            if relY < letterSize/3 && relX < letterSize*2/3 {
                if abs(relX - letterSize/3) < letterSize/6 && relY < letterSize/4 {
                    isLetter = true
                }
            }
            // Diagonal of R
            if relY > letterSize/3 && relX > letterSize/4 {
                if abs(relY - letterSize/2) < letterSize/8 {
                    isLetter = true
                }
            }
            
            if isLetter {
                pixelData[offset] = 255     // R
                pixelData[offset + 1] = 255 // G
                pixelData[offset + 2] = 255 // B
                pixelData[offset + 3] = 255 // A
            }
        }
    }
    
    let colorSpace = CGColorSpaceCreateDeviceRGB()
    let bitmapInfo = CGBitmapInfo(rawValue: CGImageAlphaInfo.premultipliedLast.rawValue)
    
    guard let context = CGContext(data: &pixelData,
                                   width: width,
                                   height: height,
                                   bitsPerComponent: bitsPerComponent,
                                   bytesPerRow: bytesPerRow,
                                   space: colorSpace,
                                   bitmapInfo: bitmapInfo.rawValue) else {
        print("Failed to create context")
        return
    }
    
    guard let cgImage = context.makeImage() else {
        print("Failed to create image")
        return
    }
    
    // Save as PNG
    let url = URL(fileURLWithPath: path)
    guard let destination = CGImageDestinationCreateWithURL(url as CFURL, UTType.png.identifier as CFString, 1, nil) else {
        print("Failed to create destination")
        return
    }
    
    CGImageDestinationAddImage(destination, cgImage, nil)
    CGImageDestinationFinalize(destination)
    
    print("✅ Created icon: \(path)")
}

// Create 1024x1024 app icon
let iconPath = "/Users/alisimacpro/Desktop/RayLink/RayLink/Assets.xcassets/AppIcon.appiconset/AppIcon.png"
createAppIcon(size: 1024, path: iconPath)